### ISSUE TITLE
PDFViewer breaks react useContext

### ISSUE THREAD

**Issue by CzBuCHi** [diegomura/react-pdf#743]
**Describe the bug**
PDFViewer component prevents passing context using react hooks.

**To Reproduce**
```ts
interface DataContext {
    data: { nested: string; } | null;
}
const defaultValue: DataContext = { data: null };
const DataContext = React.createContext(defaultValue);

// real context loaded via ajax 
const getDataContextValue = () => ({ data: { nested: "value" }});

// app root
const App: React.FC = ({ children }) => {
    return (
        <DataContext.Provider value={getDataContextValue()}>
             {children}
       </DataContext.Provider>
   );
};

// test pdf doc, that uses react hook context
const MyPdf: React.FC = () => {
    const { data: { nested } } = React.useContext(DataContext);

    return (
        <Document>
            <Page>
                <Text>{nested}</Text>
            </Page>
        </Document>
    );
};

// this will throw error during deconstructing because datacontext will have default value 
// and not value from App DataContext.Provider
const ThrowsError: React.FC = () => {
    return (
        <PDFViewer>
            <MyPdf />
        </PDFViewer>
    );
};

// this will render successfully 
const Workaroud: React.FC = () => {
    return (
        <DataContext.Consumer>
            {dataContext => (
                <PDFViewer>
                    <DataContext.Provider value={dataContext}>
                        <MyPdf />
                    </DataContext.Provider>
                </PDFViewer>
            )}
        </DataContext.Consumer>
    );
};

// render
ReactDOM.render(
    <App>
        <ThrowsError />
        <Workaroud />
    </App>
, root);
```

**Expected behavior**
PDFViewer should pass context into document without needs for  manually pass context as in Workaroud component above.

**Desktop:**
 - OS: windows 10
 - Browser chrome
 - React-pdf version 1.6.7

**Comment by fabienheureux** [comment]
I have some related issues with context as well, this may be related: https://github.com/diegomura/react-pdf/issues/522

**Comment by diegomura** [comment]
Yes. I'm aware of this and I could fix it for `v2.0` that hopefully will be ready soon. Closing this in favor of #522

**Comment by Karunika** [comment]
Any fixes?
